src/backend/distributed/operations/shard_split.c

```diff
@@ -374,14 +391,91 @@ SplitShard(SplitMode splitMode,
 							   SplitOperationName[splitOperation],
 							   SplitTargetName[splitOperation])));
 	}
-	else if (PlacementMovedUsingLogicalReplicationInTX)
+}
+
+
+/*
+ * ErrorIfMultipleNonblockingSplitInTheSameTransaction will error if we detect multiple
+ * nonblocking split operation in the same transaction.
+ */
+static void
+ErrorIfMultipleNonblockingSplitInTheSameTransaction(SplitOperation splitOperation)
+{
+	if (PlacementMovedUsingLogicalReplicationInTX)
 	{
 		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("multiple shard movements/splits via logical "
+						errmsg("Multiple '%s %s' operations via logical "
```

Comment:
postgres convention is that error messages start with lowercase  and do not have `.` at the end